File: src/main/java/spaceEngineers/SeObservation.java
package spaceEngineers;

import helperclasses.datastructures.Vec3;
import world.LegacyEntity;
import java.util.List;

public class SeObservation {

    public String agentID;
    public Vec3 position;
    public Vec3 velocity;

    public List<SeEntity> entities;

    public List<SeBlock> blocks;
}

File: Source/Ivxr.SePlugin/WorldModel/SeObservation.cs
using System;
using System.Collections.Generic;
using System.Text;
using Iv4xr.SePlugin.WorldModel;
using VRageMath;

namespace Iv4xr.SePlugin.WorldModel
{
	// Intended to be maximally compatible with Iv4xr.framework.world.WorldModel from the iv4XR Java framework.
	public class SeObservation
	{
		public string AgentID;

		public PlainVec3D Position;  // Agent's position
		public PlainVec3D OrientationForward;
		public PlainVec3D OrientationUp;
		public PlainVec3D Velocity;  // Agent's velocity
		public PlainVec3D Extent;  // Agent's dimensions (x,y,z size/2)

		public List<SeEntity> Entities;

		public List<SeBlock> Blocks;
	}
}
